src/js/components/settings/user-management/userdefinition.js

```diff
@@ -20,8 +20,15 @@ const useStyles = makeStyles()(theme => ({
   widthLimit: { maxWidth: 500 }
 }));
 
+export const getUserSSOState = user => {
+  const { sso = [] } = user;
+  const isOAuth2 = !!sso.length;
+  const provider = isOAuth2 ? OAuth2Providers.find(provider => sso.some(({ kind }) => kind.includes(provider.id))) ?? genericProvider : null;
+  return { isOAuth2, provider };
+};
+
```

Comment:
just out of curiosity, how are you dealing with unit tests? does this function have units? also: usually I would expect some test being enhanced to catch this case.

Reply:
https://northern.tech.gitlab.io/-/Mender/gui/-/jobs/3141899067/artifacts/coverage/lcov-report/js/components/settings/user-management/userdefinition.js.html#L23 <- this function seems to be called 40 times during the unit tests (since it's used in 2 components already)... although the fallback to the `genericProvider` is not caught by itself 😕, but I'll give it a go once it's on staging 😉